edit-post/store/reducer.js

```diff
@@ -49,6 +51,20 @@ export const preferences = combineReducers( {
 
 		return state;
 	},
+	disabledPanels( state = PREFERENCES_DEFAULTS.disabledPanels, action ) {
+		switch ( action.type ) {
+			case 'ENABLE_PANEL':
+				return omit( state, action.panel );
+
+			case 'DISABLE_PANEL':
+				return {
+					...state,
+					[ action.panel ]: true,
+				};
+		}
+
+		return state;
+	},
 	panels( state = PREFERENCES_DEFAULTS.panels, action ) {
```

Comment:
Wondering if I should modify the existing `panels` state tree for our needs e.g. `panels: { 'post-status': { enabled: true, shown: false } }` instead of adding a whole new `disabledPanels`. And, if so, is it worth writing migration logic so that the shown/hidden panel state isn't lost when users update Gutenberg? @talldan

Reply:
Think I'll merge `disabledPanels` and `panels`. It doesn't matter that users upgrading Gutenberg will have to re-open a few panels here and there.